src/components/node.js

```diff
@@ -178,15 +178,13 @@ function Node({
         shiftKey
       );
 
-      if (nodeUpdate && nodeUpdate.then) {
-        nodeUpdate
-          .then(() => {
-            onNodeSelected(data, data[nodeKey], shiftKey, sourceEvent);
-          })
-          .catch(() => {
-            onNodeSelected(data, data[nodeKey], shiftKey, sourceEvent);
-          });
-      }
+      nodeUpdate
+        .then(() => {
+          onNodeSelected(data, data[nodeKey], shiftKey, sourceEvent);
+        })
+        .catch(() => {
+          onNodeSelected(data, data[nodeKey], shiftKey, sourceEvent);
```

Comment:
I realize this code isn't changed as part of this diff, but what's the reason for same behavior in the case of success or failure? Since they are the same, can you share the method? For example (naming seems hard on this one, but just an idea)
```
function processAfterNodeUpdate() {
  onNodeSelected(data, data[nodeKey], shiftKey, sourceEvent);
}

nodeUpdate.then(processAfterNodeUpdate).catch(processAfterNodeUpdate);
```

Reply:
Ideally I would have used a `.finally()` call, but I can't remember if that was accepted in the EcmaScript spec recently or not, so I used both .then and .catch. We would be able to use it if we assumed everyone used Node 10 and above, or any up-to-date browser.

Reply:
Thanks I've adjusted the code based on your suggestion.